objects/src/accounts/package/mod.rs

```diff
@@ -163,19 +248,33 @@ impl ComponentMetadata {
         &self.targets
     }
 
-    pub fn storage(&self) -> &Vec<StorageEntry> {
+    pub fn storage_entries(&self) -> &Vec<StorageEntry> {
         &self.storage
     }
 }
 
 #[derive(Debug, Error)]
-pub enum ComponentMetadataError {
-    #[error("multi slot storage entry should contain one value per slot")]
+pub enum ComponentPackageError {
+    #[error("error creating AccountComponent: {0}")]
+    AccountComponentError(AccountError),
```

Comment:
Did you mean `#[source]`? I added `#[from]` in the serde error and `#[source]` here. I'd rather avoid automatically adding the conversion since there might be places where we benefit from double-checking the context